Probesys/bileto
# New/orga tree

Related to https://github.com/Probesys/bileto/issues/87

Changes proposed in this pull request:

- Add a parentsPath field to Organization entity which contains the list of parents organizations ids
- parentsPath is configured with a max depth of 3
- Add a link to the "new organization" page in the organizations cards
- In this case, a new select box is displayed to allow to select a parent organization
- Display the sub-organizations in the organizations cards
- Sort the organizations in code instead of SQL (so we handle locale-specific cases)
- Remove unique constraint on organization name (to handle the cases where two different organizations have sub-organizations with the same name)
- Display the list of organizations parents in the interface (e.g. when displaying a ticket)

How to test the feature manually:

1. create an organization, then click on "+ sub-organization"
2. create a sub-organization under the first organization
3. verify it appears on the home page in the organization card
4. create a sub-sub-organization (by attaching it to the sub-organization)
5. verify it appears on the home page in the organization card
6. verify you cannot create a sub-sub-sub-organization
7. verify you can (still) create tickets in the different organizations
8. verify the full "organizations path" appears when listing, creating or showing a ticket

Pull request checklist:


_If you think one of the item isn’t applicable to the PR, please check it
anyway and precise `N/A` next to it._
